secondhw/src/main/io/codelex/java_advanced_test/exercise3/Joiner.java

```diff
@@ -0,0 +1,24 @@
+package io.codelex.java_advanced_test.exercise3;
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+public class Joiner<T> {
+
+    private T value;
+
+    private String separator;
+
+    private Function<T, String> joiner;
+
+    public Joiner(T value, String separator, Function<T, String> joiner) {
```

Comment:
Šis tad būtu jāpārtaisa uz vienkāršāko variantu - tā lai atbilstu uzdevumam:
1. Konstruktorā ir tikai seaparator
2. join metode saņem kā parametru T... items jebšu tai var padot dažādu skaitu ar kaut kāda tipa objektiem.